### PR TITLE
Feature Add Invite Functionality

### PR DESCRIPTION
**Pull request by mojo706**
#### What does this PR do?
Add the following features:
- Invite Users
- Edit User
- Delete User
- Search User
#### How should this be manually tested?
- Go to Wire Front End and as Admin use the above listed functionality
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[159074147](https://www.pivotaltracker.com/story/show/159074147)
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/server/controllers/users.js [hunk 1]
```diff
@@ -128,5 +134,82 @@ module.exports = {
         errorLogs.catchErrors(error);
         res.status(400).send(error);
       });
+  },
+  inviteUser(req, res) {
+    const name = getUsernameFromEmail(req.body.email);
+    req.body.username = name.first + ' ' + name.last;
```
Comment (kimobrian) [inline src/server/controllers/users.js]: I will prefer using `res.locals` to store any variables local to the request. [Link](https://expressjs.com/en/api.html#res.locals)

File: src/server/controllers/users.js [hunk 2]
```diff
@@ -128,5 +134,82 @@ module.exports = {
...
+      email: req.body.email,
+      username: req.body.username,
+      roleId: req.body.roleId,
+      locationId: req.body.locationId
+    };
+    return User.findOrCreate({where: { email: userObject.email},
+      defaults: userObject})
+      .spread( async (createdUser, created) => {
+        if(!created) {
+          return res.status(409).send({ message: 'User already exists' });
```
Comment (kimobrian) [inline src/server/controllers/users.js]: If a user already exists, update their role to march the role in the request to ensure those users with the role 1 e.g reporter, witness and are PnC members who  are being made assignees or admins have their roleId updated appropriately.

File: src/server/controllers/users.js [hunk 3]
```diff
@@ -128,5 +134,82 @@ module.exports = {
...
+  },
+
+  async searchUser(req, res) {
+    try {
+      const searchQuery = { $ilike: Sequelize.fn('lower', `%${req.query.q.toLowerCase()}%`)};
+      const users = await User.findAll({
+        where: { $or: {'username': searchQuery, 'email': searchQuery}},
+        include: includes
+      });
+      return res.status(200).send({data: {users: users}, status: 'success'});
```
Comment (kimobrian) [inline src/server/controllers/users.js]: You can just write `{ users }` instead of `{users: users}`

File: src/server/helpers/emailHelper.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const smtpTransport = require('nodemailer-smtp-transport');
+
+/**
+ * @function sendMail
+ * @param Object
+ * @param function
+ */
+const sendMail = (emailBody, callback) => {
+  const transporter = nodemailer.createTransport(smtpTransport({
+    service: 'gmail',
```
Comment (kimobrian) [inline src/server/helpers/emailHelper.js]: Can we add some more to this email. Make some good HTML template for it. It's too plain

**Review by kimobrian — CHANGES_REQUESTED** [review]

File: src/server/helpers/emailHelper.js (hunk 4 above)
Comment (mojo706) [reply src/server/helpers/emailHelper.js]: Done

File: src/server/helpers/generateEmailBody.spec.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const expect = chai.expect;
+
+
+describe('#####GenerateEmailBody', () => {
+
+  it('should generate the email body', async () => {
+    const email ='eugene.omar@andela.com';
+    const roleId = 3;
+    const emailBody = await generateEmailBody(email, roleId);
+    const body = { subject: 'Invite to join WIRE',
```
Comment (kimobrian) [inline src/server/helpers/generateEmailBody.spec.js]: No idea what this body is doing though

**Pull request merged**